core/services/pipeline/task.jsonparse.go

```diff
@@ -42,12 +43,19 @@ func (t *JSONParseTask) Run(_ context.Context, _ logger.Logger, vars Vars, input
 		return Result{Error: errors.Wrap(err, "task inputs")}, runInfo
 	}
 
+	var sep StringParam
+	err = errors.Wrap(ResolveParam(&sep, From(t.Separator)), "separator")
+
 	var (
 		path JSONPathParam
 		data BytesParam
 		lax  BoolParam
 	)
-	err = multierr.Combine(
+	if sep != "" {
+		// set custom path separator
+		path = append(path, string(sep))
```

Comment:
Just wondering why `append` is used here and not just `path = []string{sep}`?

Reply:
I think b/c I had a method first before moving it inline? Updating, but it looks like it needs a bit more:
```suggestion
		path = []string{string(sep)}
```

Reply:
Actually, I think a constructor might be cleanest, rather than method or inline....